Update French locales (settings, ToS, & co)

eramdam opened the pull request:
🥐 🥖 🇫🇷  French speaking contributors rejoice 🇫🇷 🥖 🥐

~This PR adds French locales to the [landing page](https://github.com/tootsuite/mastodon/pull/4122) that was added recently and everything that came after that.~

This PR mostly add French locales to the new features added after(before?) the landing page. Mostly in the settings, as well as some confirmation modals on the web app.

If you're speaking French, feel free to comment **especially** on the `terms.body_html` string, I'm not 100% sure about the legal phrasing and speech figures to use, so if you're more knowledgeable than me on this field, I'm all ears!

~Edit: I just noticed I forgot to add locales in `fr.json`. I will do that ASAP.~ Done

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -151,15 +165,25 @@ fr:
           desc_html: Affiché sur la page d’accueil lorsque les inscriptions sont fermées<br>Vous pouvez utiliser des balises HTML
           title: Message de fermeture des inscriptions
+        deletion:
+          desc_html: Permettre à n'importe qui de supprimer leur compte
```
"n'importe qui" et "leur" ne coincident pas au plan grammatical. Je suggère "Permettre à chacun⋅e de supprimer son compte" par exemple

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -151,15 +165,25 @@ fr:
         title: Description étendue du site
+      site_terms:
+        desc_html: Vous pouvez écrire vos propres termes, conditions d'utilisations ou autre documentation juridique. Vous pouvez utiliser des tags HTML.
+        title: Termes et conditions d'utilisations personalisés
```
typo s/personalisés/personnalisées

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -194,6 +219,12 @@ fr:
   authorize_follow:
     error: Malheureusement, il y a eu une erreur en cherchant les détails du compte distant
     follow: Suivre
+    follow_request: 'Vous avez demandé⋅e à suivre:'
```
le ⋅e genré est inutile ici car le participe ne peut pas être accordé au sujet dans cette phrase

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -194,6 +219,12 @@ fr:
     error: Malheureusement, il y a eu une erreur en cherchant les détails du compte distant
     follow: Suivre
+    follow_request: 'Vous avez demandé⋅e à suivre:'
+    following: 'Youpi! Vous suivez:'
```
ajouter une espace insécable avant le deux-points

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -300,6 +331,21 @@ fr:
     prev: Précédent
+  push_notifications:
+    favourite:
+      title: "%{name} à mis⋅e votre status en favori"
```
typo s/status/statut

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -300,6 +331,21 @@ fr:
+      action_favourite: Ajouter aux favoris
+      title: "%{name} vous a mentionné⋅e"
+    reblog:
+      title: "%{name} a partagé⋅e votre status"
```
⋅e genré inutile grammaticalement comme plus haut. aussi : *statut*

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -185,6 +186,21 @@ fr:
+        nsfw_off: NSFW OFF
+        nsfw_on: NSFW ON
+      execute: Exécuter
+      failed_to_execute: Error d'exécution
```
trad oubliée Error > Erreur

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -185,6 +186,21 @@ fr:
+        show: Montrer les médias
+        title: Médias
+      no_media: Aucun média
+      title: Status du compte
```
en français c'est "statut", ou même "état"

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <p>Nous collectons des données lorsque vous vous enregistrez sur notre site et les récoltons lorsque vous participez dans le forum en lisant, écrivant, et évaluant le contenu partagé ici.</p>
+
+    <p>Lors de l'enregistrement sur notre site, il peut vous être demandé de renseigner votre nom et adresse e-mail. Vous pouvez, cependant, visiter notre site sans inscription. Votre adresse e-mail devra être vérifié grâce à un e-mail contenant un lien unique. Si ce lien est visité, nous savons que vous contrôlez cette adresse e-mail.</p>
```
typo s/vérifié/vérifiée

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <p>Lors de l'enregistrement sur notre site, il peut vous être demandé de renseigner votre nom et adresse e-mail. Vous pouvez, cependant, visiter notre site sans inscription. Votre adresse e-mail devra être vérifié grâce à un e-mail contenant un lien unique. Si ce lien est visité, nous savons que vous contrôlez cette adresse e-mail.</p>
+
+    <p>Lors de l'inscription et de la publication de status, nous enregistrons l'adresse IP de laquelle le(s) status viennent. Nous pouvons également conserver des historiques serveurs qui contiendront l'adresse IP de chaque requête adressée à notre serveur.</p>
```
statut

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <h3 id="use">Que faisons-nous avec vos données?</h3>
+
+    <p>Toute information que nous collectons pourra être utilisée d'une des manières suivantes:</p>
```
une espace insécable avant deux-points

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+    <p>Toute information que nous collectons pourra être utilisée d'une des manières suivantes:</p>
+
+    <ul>
+      <li>Pour personnaliser votre expérience &mdash; vos données nous aide à mieux répondre à vos besoins individuels.</li>
```
vos données nous aide > nous aident

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+    <ul>
+      <li>Pour personnaliser votre expérience &mdash; vos données nous aide à mieux répondre à vos besoins individuels.</li>
+      <li>Pour améliorer notre site &mdash; nous faisons tout notre possible pour améliorer notre site en fonction des données, retours et suggestions que nous recevons.</li>
+      <li>Afin d'améliorer le support client &mdash; vos données nous aide à mieux répondre à vos requêtes et demandes de support.</li>
```
aide > aident

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <h3 id="protect">Comment protégeons-nous vos données?</h3>
+      
+    <p>Nous appliquons une multitude de mesures de sécurités afin de maintenir la sécurité de vos données personnelles lorsque vous entrez, soumettez, ou accédez à vos données personnelles.</p>
```
des mesures une seule sécurité donc : "de mesures de sécurité", mais il vaudarit mieux écrire pour éviter l'inutile répétition : "Nous appliquons une multitude de mesures afin de maintenir la sécurité de vos données personnelles"

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <h3 id="cookies">Utilisons nous des "cookies"?</h3>
+
+    <p>Oui. Les cookies sont de petits fichiers qu'un site ou prestataires de services transfèrent sur le disque dur de votr ordinateur par le bais de votre navigateur Web (si ce dernier le permet). Ces cookies permettent au site de reconnaître votre navigateur et, si vous disposez d'un compte, l'associer à votre compte.</p>
```
typo s/ votr/votre

goofy-bz replied:
typo s/bais/biais

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <p>Oui. Les cookies sont de petits fichiers qu'un site ou prestataires de services transfèrent sur le disque dur de votr ordinateur par le bais de votre navigateur Web (si ce dernier le permet). Ces cookies permettent au site de reconnaître votre navigateur et, si vous disposez d'un compte, l'associer à votre compte.</p>
+
+    <p>Nous utilisons les cookies pour enregistrer vos préférences pour de futures visites, compiler des données agrégées à propos du traffic et des interactions effectuées sur le site afin de proposer une meilleure expérience dans le futur. Nous pouvons contracter les services d'acteurs tiers afin de nous aider à mieux comprendre les visiteurs de notre site. Ces acteurs n'ont l'autorisation d'ut ...
```
typo traffic > trafic
sur la fin de phrase, erreur de syntaxe
Ces acteurs n'ont > Ces acteurs ont (car "seulement" derrière suffit)

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <p>Nous utilisons les cookies pour enregistrer vos préférences pour de futures visites, compiler des données agrégées à propos du traffic et des interactions effectuées sur le site afin de proposer une meilleure expérience dans le futur. Nous pouvons contracter les services d'acteurs tiers afin de nous aider à mieux comprendre les visiteurs de notre site. Ces acteurs n'ont l'autorisation d'ut ...
+
+    <h3 id="disclose">Divulguons-nous des données à des acteurs tiers?</h3>
```
une espace insécable avant le point d'interrogation, obligatoire en français

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <h3 id="disclose">Divulguons-nous des données à des acteurs tiers?</h3>
+
+    <p>Nous n'échangeons pas, ne vendons pas ni effectuons quelconque transferts avec des acteurs tiers d'informations permettant de vous identifier personellement. Cela n'inclut pas les acteurs de confiance qui nous aident à gérer notre entreprise et à vous servir tant que ces acteurs s'accordent à garder lesdites informations confidentielles. Nous pouvons être amenés à délivrer vos informations ...
```
ajouter "de" avant "quelconque" et ajouter un S final à "quelconque"
personellement > personnellement
"aucune information permettant l'identification de nos visiteurs ne sera divulguées" > divulguée

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <h3 id="third-party">Liens vers des acteurs tiers</h3>
+
+    <p>Nous pouvons être amenés à inclure ou offrir les services ou produits d'acteurs tiers sur notre site. Ces acteurs tiers possèdent leur propre politiques de confidentialité. Nous ne sommes donc pas responsable du contenu ou activités desdits acteurs. Néanmoins, nous cherchons à protéger l'intégrité de notre site et sommes ouverts à toute remarque concernants ces acteurs.</p>
```
leur propre politiques > politique
responsable > responsables
concernants > concernant

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <h3 id="coppa">Children's Online Privacy Protection Act</h3>
+
+    <p>Notre site, nos produits et services sont tous dirigés à l'usage de personnes étant âgés de 13 ans ou plus. Si ce serveur est hébergés aux États-Unis et que vous êtes âgés de moins de 13 ans, au vu du COPPA (<a href="https://fr.wikipedia.org/wiki/Children%27s_Online_Privacy_Protection_Act">Children's Online Privacy Protection Act</a>) n'utilisez pas ce site.</p>
```
hébergés > hébergé
âgés > âgé⋅e

goofy-bz commented on config/locales/fr.yml:
```diff
@@ -432,6 +432,72 @@ fr:
+
+    <p>Si nous décidons d'apporter des changements à notre politique de confidentialité, nous les mettrons à disposition sur cette page.</p>
+
+    <p>Ce document est distributé sous licence CC-BY-SA. Il a été mis à jour pour la dernière fois le 31 Mai 2013.</p>
```
typo s/distributé/distribué
la date de 2013 ne devrait-elle pas être modifiée ?

Gargron commented on config/locales/fr.yml:
```diff
@@ -368,6 +432,72 @@ fr:
     sensitive_content: Contenu sensible
+  terms:
+    body_html: |
+    <h2>Politique de confidentialité</h2>
```
I think the syntax is

```yaml
key: |
  indented as if it's a child
  etc
```

CI seems to choke on this line. (could not find expected ':' while scanning a simple key at line 437 column 5)

ykzts commented on app/javascript/mastodon/locales/defaultMessages.json:
```diff
@@ -231,19 +231,6 @@
-  },
-  {
-    "descriptors": [
-      {
```
why delete?

ykzts commented on app/javascript/mastodon/locales/defaultMessages.json:
```diff
@@ -282,10 +269,6 @@
-        "defaultMessage": "Unfollow",
-        "id": "confirmations.unfollow.confirm"
-      },
-      {
```
why delete?

ykzts commented on app/javascript/mastodon/locales/defaultMessages.json:
```diff
@@ -298,10 +281,6 @@
-        "defaultMessage": "Are you sure you want to unfollow {name}?",
-        "id": "confirmations.unfollow.message"
-      },
-      {
```
why delete?

ykzts commented on app/javascript/mastodon/locales/fr.json:
```diff
@@ -55,8 +55,6 @@
   "confirmations.mute.confirm": "Masquer",
   "confirmations.mute.message": "Confirmez vous le masquage de {name} ?",
-  "confirmations.unfollow.confirm": "Unfollow",
-  "confirmations.unfollow.message": "Are you sure you want to unfollow {name}?",
```
why delete?

eramdam replied:
@ykzts the `yarn manage:translations` removed those strings automatically so I figured out that they must have been removed from the default locale. I can bring them back if it's an issue.

ykzts replied:
```console
$ rm -rf build
$ yarn build:development
$ yarn manage:translations
```

eramdam replied:
I'll do that, thanks!

ykzts commented on app/javascript/mastodon/locales/fr.json:
```diff
@@ -111,8 +109,8 @@
   "notifications.column_settings.mention": "Mentions :",
-  "notifications.column_settings.push": "Push notifications",
-  "notifications.column_settings.push_meta": "This device",
+  "notifications.column_settings.push": "Notifications",
```
Why does "push" remove? Here "push" is important.

eramdam commented:
@Gargron I've fixed the indenting so CI should be fixed now.


@ykzts I've put back the missing messages and re-added `push` on push notifications-related strings

@goofy-bz I've fixed most of the typos you noticed.

While updating my branch with master I've noticed @Sylvhem's PR related to the landing page was merged already. So my PR is mostly about adding missing locales. Will update the title. But @Sylvhem feel free to weigh in if you have any concerns regarding the locales I'm translating 😬
